python/pyspark/sql/connect/group.py

```diff
@@ -262,8 +264,44 @@ def applyInPandas(
 
     applyInPandas.__doc__ = PySparkGroupedData.applyInPandas.__doc__
 
-    def applyInPandasWithState(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("applyInPandasWithState() is not implemented.")
+    def applyInPandasWithState(
```

Comment:
Minor: Could you remove `assert state.get[0] == 1` in the code example in the PR description? I think it is not required.